### PR TITLE
organisation impl ready for #sb-10811

### PR DESCRIPTION
**Pull request by iostream04**

### CODE REVIEW

File: samza/jobs-common/src/main/java/org/sunbird/validator/MessageValidator.java [hunk 1]
```diff
@@ -5,24 +5,24 @@
...
+
+    validate(message, Constants.IDENTIFIER, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.OBJECT_TYPE, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.OPERATION_TYPE, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.EVENT, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.EVENT_TYPE, ResponseCode.mandatoryParamsMissing);
+  }
+
+  public void validate(Map<String, Object> message, String attribute, ResponseCode responseCode) {
+    if (!message.containsKey(attribute)) {
```
Comment (bvinayakumar) [inline samza/jobs-common/src/main/java/org/sunbird/validator/MessageValidator.java]: Add isNotBlank check.

File: samza/jobs-common/src/main/java/org/sunbird/validator/MessageValidator.java [hunk 2]
```diff
@@ -5,24 +5,24 @@
...
-      ProjectCommonException.throwClientErrorException(ResponseCode.mandatoryParamsMissing, Constants.OBJECT_TYPE);
-    } else if (!message.containsKey(Constants.OPERATION_TYPE)) {
-      ProjectCommonException.throwClientErrorException(ResponseCode.mandatoryParamsMissing, Constants.OPERATION_TYPE);
-    } else if (!message.containsKey(Constants.EVENT)) {
-      ProjectCommonException.throwClientErrorException(ResponseCode.mandatoryParamsMissing, Constants.EVENT);
-    } else if (!message.containsKey(Constants.IDENTIFIER)) {
+
+    validate(message, Constants.IDENTIFIER, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.OBJECT_TYPE, ResponseCode.mandatoryParamsMissing);
+    validate(message, Constants.OPERATION_TYPE, ResponseCode.mandatoryParamsMissing);
```
Comment (bvinayakumar) [inline samza/jobs-common/src/main/java/org/sunbird/validator/MessageValidator.java:22]: Validate supported operation (UPSERT, DELETE) and event (transactional, instructional) types.

File: samza/jobs-common/src/main/java/org/sunbird/models/Constants.java [hunk 3]
```diff
@@ -20,5 +20,7 @@
   public static final String INDEX = "index";
   public static final String DOT = ".";
   public static final String TRANSACTIONAL = "transactional";
```
Comment (bvinayakumar) [inline samza/jobs-common/src/main/java/org/sunbird/models/Constants.java]: TRANSACTIONAL should be part of Message class.

File: samza/indexer/src/main/java/org/sunbird/jobs/samza/service/IndexerService.java [hunk 4]
```diff
@@ -67,10 +74,33 @@ private void updateES(String type, String key, Message message) {
...
+      data = prepareOrgData(message.getProperties());
+    } else {
+      data = message.getProperties();
+    }
     return data;
   }
 
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> prepareOrgData(Map<String, Object> map) {
+    String contactDetails = (String) map.get(JsonKey.CONTACT_DETAILS);
```
Comment (bvinayakumar) [inline samza/indexer/src/main/java/org/sunbird/jobs/samza/service/IndexerService.java]: Let us discuss regarding CONTACT_DETAILS. Perhaps, we can drop this from org metadata.

Comment (iostream04) [reply samza/indexer/src/main/java/org/sunbird/jobs/samza/service/IndexerService.java]: okay

**Pull request merged**